docs/tutorial/from-zero-to-hero-write-your-first-kubernetes-charm/create-a-minimal-kubernetes-charm.md

```diff
@@ -417,6 +417,304 @@ In the output you should see the definition for both containers. You'll be able
 
 **Congratulations, you've successfully created a minimal Kubernetes charm!** 
 
+(write-unit-tests-for-your-charm)=
+## Write unit tests for your charm
+
+When you're writing a charm, you will want to ensure that it will behave reliably as intended.
+
+For example, that the various components -- relation data, Pebble services, or configuration files -- all behave as expected in response to an event.
+
+You can ensure all this by writing a rich battery of unit tests. In the context of a charm we recommended using [`pytest`](https://pytest.org/) (but [`unittest`](https://docs.python.org/3/library/unittest.html) can also be used) and especially the operator framework's built-in testing library --  [](ops_testing_harness). We will be using the Python testing tool [`tox`](https://tox.wiki/en/4.14.2/index.html) to automate our testing and set up our testing environment.
+
+<!-- TODO
+
+This chapter and the next should be removed and the content spread throughout
+each of the previous chapters. Each time a feature is added, a unit and
+integration test should also be added. At the end of each chapter, before
+manually checking the feature works, the user should run the tests to make sure
+that they pass.
+
+-->
+
```

Comment:
We can remove this now. It ended up sitting in my PR for ages and then not very long in main, oh well.